From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Fri, 8 Sep 2023 13:25:14 +0800
Subject: cogl/onscreen-glx: Forcefully disable
 cogl_onscreen_glx_get_buffer_age

This restores the pre-45.rc behaviour of the function always failing
gracefully. It's a workaround while we agree on the best long term
solution.

The regression occurred in 55e3b2e5 from mutter!3230.

Bug: https://gitlab.gnome.org/GNOME/mutter/-/issues/3007
Bug-Ubuntu: https://launchpad.net/bugs/2034672
---
 cogl/cogl/winsys/cogl-onscreen-glx.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cogl/cogl/winsys/cogl-onscreen-glx.c b/cogl/cogl/winsys/cogl-onscreen-glx.c
index 7c9b4df..19726c8 100644
--- a/cogl/cogl/winsys/cogl-onscreen-glx.c
+++ b/cogl/cogl/winsys/cogl-onscreen-glx.c
@@ -541,6 +541,8 @@ cogl_onscreen_glx_get_buffer_age (CoglOnscreen *onscreen)
   GLXDrawable drawable;
   unsigned int age = 0;
 
+  return 0;
+
   if (!_cogl_winsys_has_feature (COGL_WINSYS_FEATURE_BUFFER_AGE))
     return 0;
 
